From b2d0308f4769b529dbeebcc94f7119a2ac5eb637 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Fri, 11 Mar 2022 22:01:10 +0100
Subject: [PATCH 083/469] input: cyttsp4: Fix probe oops

__set_bit is called with bit offset -1 which causes oops.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/input/touchscreen/cyttsp4_core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/input/touchscreen/cyttsp4_core.c b/drivers/input/touchscreen/cyttsp4_core.c
index ba0cdd3aa9d0..5d16e085ef42 100644
--- a/drivers/input/touchscreen/cyttsp4_core.c
+++ b/drivers/input/touchscreen/cyttsp4_core.c
@@ -1903,6 +1903,9 @@ static int cyttsp4_setup_input_device(struct cyttsp4 *cd)
 	for (i = 0; i < cd->n_signals; i++) {
 		signal = &cd->signals[i];
 
+		if (signal->signal < 0)
+			continue;
+
 		__set_bit(signal->signal, cd->input->absbit);
 
 		max = signal->max;
-- 
2.34.1

